### ISSUE TITLE
The class_alias in bootstrap.php causes warnings when using preloading

### ISSUE THREAD

**Issue by hussainweb** [Chi-teck/drupal-code-generator#47]
I am using a generated preloading script which looks something like this:

```php
$loader = require_once '/app/web/autoload.php';
// more lines to load PSR-4 directories (irrelevant to this issue)

$files = [
    '/app/vendor/autoload.php',
    '/app/load.environment.php',
    '/app/vendor/composer/autoload_static.php',
    // more files to be preloaded
];

foreach ($files as $file) {
    try {
        if (!(is_file($file) && is_readable($file))) {
            throw new \Exception("{$file} does not exist or is unreadable.");
        }
        require_once $file;
    } catch (\Throwable $e) {
        // ...
        throw $e;
    }
}
```

This script is set to [`opcache.preload`](https://www.php.net/manual/en/opcache.configuration.php#ini.opcache.preload) setting. The script works and I see that the preload is making a difference, but there is a warning on all pages.

```

Warning: Cannot declare class DrupalCodeGenerator\Twig\TwigEnvironment, because the name is already in use in /app/vendor/chi-teck/drupal-code-generator/src/bootstrap.php on line 44

Warning: Cannot declare class DrupalCodeGenerator\TwigEnvironment, because the name is already in use in /app/vendor/chi-teck/drupal-code-generator/src/bootstrap.php on line 47
```

After debugging, I realized this is because bootstrap.php is loaded by composer (it's part of composer.json's `autoload.files` option). As the warnings suggest, this is due to the `class_alias` line.

After a lot of debugging and research, I conclude that the problem is that the preloading works somewhat differently for `class_alias`. In other words, this could be a PHP bug and I do some hints for this in https://github.com/php/php-src/pull/3538 and https://github.com/symfony/symfony/issues/29105.

My theory is that while preloading correctly handles all the regular classes and functions, it cannot handle class_alias properly. Since class_alias runs both while preloading (as we are requiring the autoloader) and during the execution, it seems that the internal PHP entry for the class name persists during the runtime causing the warning.

The fix in my case was to wrap the class_alias call inside an if block with `class_exists` condition. I don't see any warnings after that. I'm going to create a PR against 1.x branch for that shortly.

Footnote: This is tangentially related to #46 but I don't know if this change would help static analyzers.

**Comment by Chi-teck** [comment]
What is your PHP version?
PHP prior to version 7.4.2 had some issues with preloading.

**Comment by hussainweb** [comment]
@Chi-teck, it's PHP 7.4.8.

**Comment by hussainweb** [comment]
Also, do you remember what kind of issues were there before PHP 7.4.2? I could use that for the documentation for the module I am working on. Thanks! :)

**Comment by Chi-teck** [comment]
I am not quite sure. I've just found this quote from @nikic
> I should probably clarify that 7.4.1 will not contain all the preloading fixes. Some of them will only make 7.4.2

The PHP release notes contain some useful references. Try to search  "preloading" on that page.
https://www.php.net/ChangeLog-7.php#PHP_7_4

**Comment by hussainweb** [comment]
Thanks! I'll include all that in the documentation. Please let me know if I can help here in another way.

**Comment by Chi-teck** [comment]
Fixed in #48. Thank you.

**Comment by hussainweb** [comment]
Thank you for merging this and also creating a release. It makes my documentation much easier. :)
